### DISCUSSION TITLE
4.3.4 스프링 시큐리티 설정 질문입니다.

### DISCUSSION THREAD

**Discussion by DoriJJong** [fsoftwareengineer/todo-application#99] [질문과 답변] [answered]
**request** : localhost:8080/auth/signin
**response** : 
{
    "token": "eyJhbGciOiJIUzUxMiJ9.eyJzdWIiOiI5ZjdmZGYzMS1mMzAzLTQ1YWQtOGMzOS1jYzZhMzkwZGE0MzIiLCJpc3MiOiJGaWxtaW5nIiwiaWF0IjoxNjQ3NTg1Njc2LCJleHAiOjE2NDc2NzIwNzZ9.JlS14rDS10qukvR87HVt097muQRQjlxnZ8BG6Yj2uuvu8eAFGGJ-H_SWToa9Y0pOthk308Iz4mBD_XgOD2N4dw",
    "email": "test4@test.com",
    "nickName": null,
    "password": null,
    "uuid": "9f7fdf31-f303-45ad-8c39-cc6a390da432"
}

**request** : localhost:8080/test
**response** : 
{
    "timestamp": "2022-03-18T06:41:42.376+00:00",
    "status": 403,
    "error": "Forbidden",
    "path": "/test"
}

public class TestController {

    @GetMapping("/test")
    public String test() {
        return "test";
    }
}

위와 같이 서블릿 필터 적용 후 테스트 컨트롤러로 테스트 시 에러가 발생합니다.
에러 내용은 io.jsonwebtoken.security.WeakKeyException: The verification key's size is 480 bits which is not secure enough for the HS512 algorithm. 입니다.
로깅해보니 validateAndGetUserId 에서 발생하는 에러로 확인됩니다.
(**https://github.com/fsoftwareengineer/todo-application/discussions/48 참고하여 키 관련해서는 변경**)

이전 실습에서는 해당 에러가 발생하지 않는데 무슨 차이인지 알 수 있을까요?
더 필요하신 정보가 있다면 말씀 부탁 드립니다!

**Comment by fsoftwareengineer** [comment, marked as answer]
라이브러리가 업데이트 되면서 변경된 것으로 생각합니다. 치명적인 보안 이슈의 경우 버전 업그레이드 없이도 보안패치가 가능하다는 점을 유의 해 주시기바랍니다.
